pkg/model/deployment_source.proto

```diff
@@ -0,0 +1,45 @@
+// Copyright 2024 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package model;
+option go_package = "github.com/pipe-cd/pipecd/pkg/model";
+
+message DeploymentSource {
+  // The application directory where the source code is located.
+  string application_directory = 1;
+  // The git commit revision of the source code.
+  string revision = 2;
+  // The configuration of the application which is independent for plugins.
+  GenericApplicationSpec generic_application_config = 3;
+  // The configuration of the application which is specific for plugins.
+  PluginApplicationSpec application_config = 4; 
+}
+
+message GenericApplicationSpec {
+  // TODO: Add more fields to support generic application spec.
```

Comment:
![Screenshot 2024-08-06 at 11 15 31](https://github.com/user-attachments/assets/1b1170b5-5863-4e40-b46b-6d299f13c57c)
I just realized the plugin does not need the GenericApplicationSpec at all 🤔

Reply:
I removed GenericApplicationSpec from the model because the plugin does not need it, as you pointed out.